cachito/web/content_manifest.py

```diff
@@ -232,3 +238,205 @@ def generate_icm(self, image_contents=None):
         icm = deepcopy(BASE_ICM)
         icm["image_contents"] = image_contents or []
         return deep_sort_icm(icm)
+
+
+@dataclass(frozen=True)
+class Package:
+    """
+    A package within a content manifest.
+
+    It is used primarily to generate a package URL (purl).
+    """
+
+    name: str
+    type: str
+    version: str
+    dev: bool = False
+    dependencies: list = field(default_factory=list)
+    path: str = None
+    replaces: str = None
+
+    # def __post_init__(self):
+    #     if self.dependencies:
+    #         dependencies = [Package.from_json(dependency) for dependency in self.dependencies]
+    #     else:
+    #         dependencies = []
+    #
+    #     object.__setattr__(self, "dependencies", dependencies)
+
+    def __repr__(self):
+        return (
+            f"<{self.__class__.__name__} name={self.name}, type={self.type}, "
+            f"version={self.version}>"
+        )
+
+    def __hash__(self):
+        return hash((self.name, self.type, self.version, self.dev))
+
+    @classmethod
+    def from_json(cls, package):
+        """
+        Create a Package object from JSON.
+
+        All dependencies will also be converted to Package objects.
+
+        :param dict package: the dictionary representing the package
+        :return: the Package object
+        :rtype: Package
+        """
+        # Avoids changing the original JSON dependencies
+        package = package.copy()
+
+        if "dependencies" in package:
+            package["dependencies"] = [
+                Package.from_json(dependency) for dependency in package["dependencies"]
+            ]
+
+        return cls(**package)
```

Comment:
Since we took away the `replaces` attribute, I had to rewrite the whole thing so we wouldn't have errors on the __init__ when this key was actually available in the json package.

I did avoid copying the dict, anyway :+1: